rueidisaside/typed_aside.go

```diff
@@ -0,0 +1,74 @@
+package rueidisaside
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+// TypedCacheAsideClient is an interface that provides a typed cache-aside client.
+// It allows you to cache and retrieve values of a specific type T.
+type TypedCacheAsideClient[T any] interface {
+	Get(ctx context.Context, ttl time.Duration, key string, fn func(ctx context.Context, key string) (val *T, err error)) (val *T, err error)
+	Del(ctx context.Context, key string) error
+	Client() CacheAsideClient
+}
+
+// typedCacheAsideClient is an implementation of the TypedCacheAsideClient interface.
+// It provides a typed cache-aside client that allows caching and retrieving values of a specific type T.
+type typedCacheAsideClient[T any] struct {
+	client       CacheAsideClient
+	serializer   func(*T) (string, error)
+	deserializer func(string) (*T, error)
+}
+
+// NewTypedCacheAsideClient creates a new TypedCacheAsideClient instance that provides a typed cache-aside client.
+// The client, serializer, and deserializer functions are used to interact with the underlying cache.
+// The serializer function is used to convert the provided value of type T to a string, and the deserializer function
+// is used to convert the cached string value back to the original type T.
+func NewTypedCacheAsideClient[T any](
+	client CacheAsideClient,
+	serializer func(*T) (string, error),
+	deserializer func(string) (*T, error),
+) (TypedCacheAsideClient[T], error) {
+	return &typedCacheAsideClient[T]{
+		client:       client,
+		serializer:   serializer,
+		deserializer: deserializer,
+	}, nil
+}
+
+// Get retrieves a value of type T from the cache, or fetches it using the provided
+// function and stores it in the cache. The value is cached for the specified TTL.
+// If the value cannot be retrieved or deserialized, an error is returned.
+func (c typedCacheAsideClient[T]) Get(ctx context.Context, ttl time.Duration, key string, fn func(ctx context.Context, key string) (val *T, err error)) (val *T, err error) {
+	strVal, err := c.client.Get(ctx, ttl, key, func(ctx context.Context, key string) (val string, err error) {
+		result, err := fn(ctx, key)
+		if err != nil {
+			return "", fmt.Errorf("get value by key %s failed: %w", key, err)
```

Comment:
```suggestion
			return "", err
```

I think we don't need to wrap the error here.

Reply:
[`eaa6814` (#717)](https://github.com/redis/rueidis/pull/717/commits/eaa681438c7f0ac4ce77bff5c05670181ce0435a)

Reply:
[`eaa6814` (#717)](https://github.com/redis/rueidis/pull/717/commits/eaa681438c7f0ac4ce77bff5c05670181ce0435a)